appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java

```diff
@@ -56,12 +56,12 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
                         if (environment != null) {
                             DescriptorEnvironmentLifecycleMethodProcessor.handleMethods(environment, moduleDescription, mainClass);
                         }
-                    } catch (ClassNotFoundException e) {
+                    } catch (Throwable e) {
```

Comment:
This changes the error handling for L50-58.

Refactoring a bit avoids that:

```
                   final Class<?> clazz;
                    try {
                        clazz = module.getClassLoader().loadClass(mainClass);
                    } catch (Throwable t) {
                        throw AppClientLogger.ROOT_LOGGER.cannotLoadAppClientMainClass(t);
                    }
                   deploymentUnit.putAttachment(AppClientAttachments.MAIN_CLASS, clazz);
                   ...
```

Reply:
Thanks, looks better!